src/main/java/org/commcare/session/RemoteQuerySessionManager.java

```diff
@@ -258,35 +258,26 @@ public void refreshItemSetChoices() {
     public void refreshInputDependentState() {
         refreshItemSetChoices();
         validateUserAnswers();
-        recalculateRequired();
     }
 
-    private void recalculateRequired() {
-        requiredPrompts = new Hashtable<>();
-        OrderedHashtable<String, QueryPrompt> userInputDisplays = getNeededUserInputDisplays();
-        EvaluationContext ec = getEvaluationContextWithUserInputInstance();
-        for (Enumeration en = userInputDisplays.keys(); en.hasMoreElements(); ) {
-            String key = (String)en.nextElement();
-            QueryPrompt queryPrompt = userInputDisplays.get(key);
-            boolean isRequired = queryPrompt.isRequired(ec);
-            requiredPrompts.put(key, isRequired);
-        }
-    }
 
     private void validateUserAnswers() {
+        requiredPrompts = new Hashtable<>();
         errors = new Hashtable<>();
         OrderedHashtable<String, QueryPrompt> userInputDisplays = getNeededUserInputDisplays();
         String instanceId = VirtualInstances.makeSearchInputInstanceID(getSearchInstanceReferenceId());
         EvaluationContext ec = getEvaluationContextWithUserInputInstance();
         for (Enumeration en = userInputDisplays.keys(); en.hasMoreElements(); ) {
             String key = (String)en.nextElement();
             QueryPrompt queryPrompt = userInputDisplays.get(key);
+            boolean isRequired = queryPrompt.isRequired(ec);
             String value = userAnswers.get(key);
+            requiredPrompts.put(key, isRequired);
```

Comment:
lint: probably move the line one up to be in continuation with `isRequired` logic